reports/helmcharts/helm-chart-versions.sh

```diff
@@ -130,21 +129,53 @@ for chart in $(echo "$charts" | jq -c '.[]'); do
   fi
 
   # Enhance document with additional information
+  helm_chart_name=$(echo "$chart" | jq -r '.name')
   uuid=$(uuidgen)
   created_on=$(date '+%Y-%m-%d %H:%M:%S')
+  id="${cluster_name}-${helm_chart_name}"
+
 
   document=$(echo "$chart" | jq --arg cluster_name "$cluster_name" \
                                 --arg verdict $verdict \
-                                --arg id "$uuid" \
+                                --arg id "$id" \
                                 --arg environment "$environment" \
                                 --arg created_on "$created_on" \
                                 --arg report_type "table" \
                                 --arg display_name "HELM Repositories" \
                                 --arg color_code $color_code '. + {id: $id, environment: $environment, createdOn: $created_on, lastUpdated: $created_on, displayName: $display_name, cluster: $cluster_name, verdict: $verdict, colorCode: $color_code, reportType: $report_type}')
 
   documents+=("$document")
+  
 done
 
+cosmosdb_account_name="$COSMOSDB_ACCOUNT_NAME"
+cosmosdb_database_name="$COSMOS_DB_NAME"
+cosmosdb_container_name="$COSMOS_DB_CONTAINER"
+id_to_check="$id"
+
+query_result=$(az cosmosdb sql container execute-query \
+--account-name "$cosmosdb_account_name" \
+--database-name "$cosmosdb_database_name" \
+--name "$cosmosdb_container_name" \
+--query "SELECT * FROM c WHERE c.id = '$id_to_check'" \
+--output json)
+
+if [[ $query_result == "[]" ]]; then
```

Comment:
I'm confused how this works. 
The save-to-cosmos.py removes all documents then adds all new ones for each run of the script.

Your query exists outside the for loop so how does it check every single $id as that value should change for each chart found and each run of the for loop: `for chart in $(echo "$charts" | jq -c '.[]'); do`